feature/gnoi/factory_reset/factory_reset_test.go

```diff
@@ -0,0 +1,206 @@
+		}
+
+		filesCreated = append(filesCreated, fPath)
+	}
+	for _, fP := range filesCreated {
+		resp, err := cli.SendCommand(context.Background(), fmt.Sprintf(checkFileExists, fP))
+		if err != nil {
+			t.Fatalf("Failed to send command %s on the device, Error: %v", fmt.Sprintf(checkFileExists, fP), err)
+		}
+		t.Logf("%v", resp)
+		if !strings.Contains(resp, fileExists) {
+			t.Fatalf("Unable to Create a file object %s in device %s", fP, dut.Name())
+		}
+	}
+
+}
+
+// checkFiles check if the files created are deleted from the device after factory reset
+func checkFiles(t *testing.T, dut *ondatra.DUTDevice) {
+	for _, fP := range filesCreated {
+
+		resp, err := dut.RawAPIs().CLI(t).SendCommand(context.Background(), fmt.Sprintf(checkFileExists, fP))
+		if err != nil {
+			t.Fatalf("Failed to send command %s on the device, Error: %v", fmt.Sprintf(checkFileExists, fP), err)
+		}
+		t.Logf(resp)
+		if strings.Contains(resp, fileExists) == true {
+			t.Fatalf("File %s not cleared by system Reset, in device %s", fP, dut.Name())
+		}
+
+	}
+}
+
+func deviceBootStatus(t *testing.T, dut *ondatra.DUTDevice) {
+	startReboot := time.Now()
+	t.Logf("Wait for DUT to boot up by polling the telemetry output.")
+	for {
+		var currentTime string
+		t.Logf("Time elapsed %.2f minutes since reboot started.", time.Since(startReboot).Minutes())
+
+		time.Sleep(3 * time.Minute)
+		if errMsg := testt.CaptureFatal(t, func(t testing.TB) {
+			currentTime = dut.Telemetry().System().CurrentDatetime().Get(t)
+		}); errMsg != nil {
+			t.Logf("Got testt.CaptureFatal errMsg: %s, keep polling ...", *errMsg)
+		} else {
+			t.Logf("Device rebooted successfully with received time: %v", currentTime)
+			break
+		}
+
+		if uint64(time.Since(startReboot).Minutes()) > maxRebootTime {
+			t.Fatalf("Check boot time: got %v, want < %v", time.Since(startReboot), maxRebootTime)
+		}
+	}
+	t.Logf("Device boot time: %.2f minutes", time.Since(startReboot).Minutes())
+}
+
+// performs factory reset
+func factoryReset(t *testing.T, dut *ondatra.DUTDevice, devicePaths []string) {
+	createFiles(t, dut, devicePaths)
+	gnoiClient := dut.RawAPIs().GNOI().New(t)
+	facRe, err := gnoiClient.FactoryReset().Start(context.Background(), &frpb.StartRequest{FactoryOs: false, ZeroFill: false})
+	if err != nil {
+		t.Fatalf("Failed to initiate Factory Reset on the device, Error : %v ", err)
+	}
+	t.Logf("Factory reset Response %v ", facRe)
+	time.Sleep(2 * time.Minute)
+	deviceBootStatus(t, dut)
+	dutNew := ondatra.DUT(t, "dut")
+	checkFiles(t, dutNew)
+	t.Log("Factory reset successfull")
+}
+
+func TestFactoryReset(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+
+	switch dut.Vendor() {
+	case ondatra.CISCO:
+		enCiscoCommands = encryptionCommands{EncrytionStatus: "show disk-encryption status", EncryptionActivate: "disk-encryption activate", EncryptionDeactivate: "disk-encryption deactivate", DevicePaths: []string{"/misc/disk1"}}
```

Comment:
@xw-g @sachendras @vcheny  we need to agree internally for the CLI usage in our tests.